pandas/tests/groupby/test_function.py

```diff
@@ -1590,11 +1590,11 @@ def test_corrwith_with_1_axis():
     tm.assert_series_equal(result, expected)
 
 
-@pytest.mark.filterwarnings("ignore:The 'mad' method.*:FutureWarning")
+@pytest.mark.filterwarnings("ignore:.* is deprecated:FutureWarning")
 def test_multiindex_group_all_columns_when_empty(groupby_func):
     # GH 32464
     df = DataFrame({"a": [], "b": [], "c": []}).set_index(["a", "b", "c"])
-    gb = df.groupby(["a", "b", "c"])
+    gb = df.groupby(["a", "b", "c"], group_keys=False)
```

Comment:
I added 

```
diff --git a/pandas/core/groupby/groupby.py b/pandas/core/groupby/groupby.py
index 16ee154156..8d442c6ae3 100644
--- a/pandas/core/groupby/groupby.py
+++ b/pandas/core/groupby/groupby.py
@@ -1040,7 +1040,7 @@ class GroupBy(BaseGroupBy[NDFrameT]):
                 return self._obj_with_exclusions

             result = self._python_apply_general(
-                curried, self._obj_with_exclusions, is_transform=is_transform
+                curried, self._obj_with_exclusions, is_transform=is_transform, not_indexed_same=not is_transform
             )

             if self._selected_obj.ndim != 1 and self.axis != 1 and result.ndim != 1:
```

and still get

```
pandas/tests/groupby/test_function.py::test_multiindex_group_all_columns_when_empty[idxmax]
pandas/tests/groupby/test_function.py::test_multiindex_group_all_columns_when_empty[idxmin]
  .../pandas/tests/groupby/test_function.py:1601: FutureWarning: Not prepending group keys to the result index of transform-like apply. In the future, the group keys will be included in the index, regardless of whether the applied function returns a like-indexed object.
  To preserve the previous behavior, use

  	>>> .groupby(..., group_keys=False)

  To adopt the future behavior and silence this warning, use

  	>>> .groupby(..., group_keys=True)
    result = method(*args).index
```

Ignoring the warnings at this line also still raises this warnings for idxmin/idxmax so that must take a different code path?

Reply:
Ahh, that's right; they are listed in common_apply_allowlist but then also defined on `DataFrameGroupBy` (I have a PR for #48028 that does away with common_apply_allowlist completely that I'll put up once 1.5 is released).

The same can be done with idxmin/max on DataFrameGroupBy.